.gitlab-ci.yml

```diff
@@ -0,0 +1,118 @@
+before_script:
+  # set stack isolation
+  - export ISOLATION=buildpipeline${CI_PIPELINE_ID}${CI_BUILD_NAME}
+  - export COMPOSE_PROJECT_NAME=${ISOLATION}
+  - export TUPLE_C=$(expr ${CI_BUILD_ID} % 99)
+  - echo ${TUPLE_C}
+  # run docker-compose commands from tests environment
+  - cd tests
+  - cp .env-dist .env
+  - docker-compose config
+
+after_script:
+  - export ISOLATION=buildpipeline${CI_PIPELINE_ID}${CI_BUILD_NAME}
+  - export COMPOSE_PROJECT_NAME=${ISOLATION}
+  # run docker-compose commands from tests environment
+  - cd tests
+  - cp .env-dist .env
+  - docker-compose down -v --remove-orphans
+  - docker ps -f name=${ISOLATION}
+
+stages:
+  - travis
+  - test
+  - cleanup
+
+test:
+  stage: test
+  script:
+    - docker-compose up --build -d
+    - docker-compose run --rm php vendor/bin/phpunit -v --exclude caching,db,data --log-junit tests/_junit/test.xml
```

Comment:
starting in the background is `-d`, removing the container after it has been stopped is `--rm`.